hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/RDBSnapshotProvider.java

```diff
@@ -110,21 +111,25 @@ public DBCheckpoint downloadDBSnapshotFromLeader(String leaderNodeID)
         "reloading state from the snapshot.", leaderNodeID);
     checkLeaderConsistency(leaderNodeID);
 
-    String snapshotFileName = getSnapshotFileName(leaderNodeID);
-    File targetFile = new File(snapshotDir, snapshotFileName);
-    downloadSnapshot(leaderNodeID, targetFile);
-    LOG.info("Successfully download the latest snapshot {} from leader OM: {}",
-        targetFile, leaderNodeID);
-
-    numDownloaded.incrementAndGet();
-    injectPause();
-
-    RocksDBCheckpoint checkpoint = getCheckpointFromSnapshotFile(targetFile,
-        candidateDir, true);
-    LOG.info("Successfully untar the downloaded snapshot {} at {}.", targetFile,
-        checkpoint.getCheckpointLocation());
-
-    return checkpoint;
+    while (true) {
+      String snapshotFileName = getSnapshotFileName(leaderNodeID);
+      File targetFile = new File(snapshotDir, snapshotFileName);
+      downloadSnapshot(leaderNodeID, targetFile);
+      LOG.info(
+          "Successfully download the latest snapshot {} from leader OM: {}",
+          targetFile, leaderNodeID);
+
+      numDownloaded.incrementAndGet();
+      injectPause();
+
+      RocksDBCheckpoint checkpoint = getCheckpointFromSnapshotFile(targetFile,
+          candidateDir, true);
+      if (ratisSnapshotComplete(checkpoint.getCheckpointLocation())) {
+        LOG.info("Successfully untar the downloaded snapshot {} at {}.",
+            targetFile, checkpoint.getCheckpointLocation());
+        return checkpoint;
```

Comment:
The log message here IMO is not correct. The LOG inside the condition should be moved out. And add extra logging if uncessary:

```suggestion
      RocksDBCheckpoint checkpoint = getCheckpointFromSnapshotFile(targetFile,
          candidateDir, true);
      LOG.info("Successfully untar the downloaded snapshot {} at {}.",
          targetFile, checkpoint.getCheckpointLocation());
      if (ratisSnapshotComplete(checkpoint.getCheckpointLocation())) {
        LOG.info("Ratis snapshot transfer is complete.");
        return checkpoint;
```

Reply:
Best if we could keep a counter for the loop, in order to log the tarball sequence number for debugging?